src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/UniversityTest.java

```diff
@@ -0,0 +1,100 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class UniversityTest {
+    static List<Student> students = Arrays.asList(
+            new Student("Zhenya", "Evmakov", "KIPIA", 5),
+            new Student("Asya", "Asina", "Chemistry", 3),
+            new Student("A", "A", "KIPIA", 5),
+            new Student("B", "B", "KIPIA", 1),
+            new Student("C", "C", "Chemistry", 3));
+
+    @Test
+    static Stream<Arguments> groupingByCourseTestProviderArguments() {
+        return Stream.of(
+                Arguments.of(Map.of(
+                        1, Arrays.asList(students.get(3)),
+                        3, Arrays.asList(students.get(1), students.get(4)),
+                        5, Arrays.asList(students.get(0), students.get(2))))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("groupingByCourseTestProviderArguments")
+    public void groupingByCourseTest(Map<Integer, List<Student>> expected) {
+        Map<Integer, List<Student>> actual = University.groupingByCourse(students);
+        assertEquals(expected, actual);
```

Comment:
"Violation: There isn't separating empty lines for GIVEN, WHEN, THEN parts
Solution: Add blank line"